src/env/to_envp.c

```diff
@@ -0,0 +1,42 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*   tosh - 21 Shell                                      ::::::::            */
+/*                                                      :+:    :+:            */
+/*   By: aholster <aholster@student.codam.nl>          +:+                    */
+/*       ivan-tey <ivan-tey@student.codam.nl>         +#+                     */
+/*       nloomans <nloomans@student.codam.nl>        +#+                      */
+/*                                                 #+#    #+#                 */
+/*   License: GPLv3                                ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "private.h"
+#include "ft_printf.h"
+
+char					**env_to_envp(t_env *const env)
+{
+	size_t				i;
+	t_list_conn			*iter;
+	struct s_env_pair	*temp;
+	char				**env_arr;
+
+	i = 0;
+	iter = env->list.first;
+	env_arr = ft_memalloc(sizeof(char *) * (env->list.len + 1));
+	if (env_arr == NULL)
+		return (NULL);
+	while (i < env->list.len && iter)
+	{
+		temp = unpack_env(iter);
+		ft_asprintf(&env_arr[i], "%s=%s", temp->key, temp->value);
+		if (env_arr[i] == NULL)
+		{
+			ft_arraydel((void ***)&env_arr, ft_memdel);
+			return (NULL);
+		}
+		iter = iter->next;
+		i++;
+	}
```

Comment:
add assert to assure `list.len` matches actual `len` rather than a double check as while condition

Reply:
```
while (i < env->list.len)
{
    assert(iter != NULL);
    temp = unpack_env(iter);
```